AEROGEAR-3332 - Cordova build errors

ciaranRoche opened the pull request:
## Motivation

<!-- The reason underlying the contents of the PR, can be a link to the originating JIRA -->

JIRA: https://issues.jboss.org/browse/AEROGEAR-3327

## Description

Currently, we cannot build Cordova showcase App if it was previously built for a different platform, this appears to be down to conflicting files which are specific to each platform.
This change adds a clean script to be run between different platform builds along with an update to the documentation and build steps.

## Progress

- [x] Add Script
- [x] Update Docs

## Verification Steps

- From a clean project run `npm run ionic:android`
- Once built, run `npm run clean`
- Next run `npm run ionic:ios`
(Both builds should complete successfully.

psturc commented:
👀

wei-lee commented:
@ciaranRoche Not sure if this is related to your change but I can get `npm run ionic:ios` and `npm run ionic:android` working. I am getting the errors [here](https://gist.github.com/wei-lee/26e2024aab3072f3668dc3c0453d5a7e).

I did ran `npm run clean` before each build but still endup with the same error. Any idea?

ciaranRoche commented:
@wei-lee only difference I see is you are using node `v6.10.3` as far as I am aware ionic kicks up if you are not using `v8.10.0`, Id say bump your node version and see if it makes a difference if it does fix it we should capture it in the docs.

ciaranRoche commented:
@wei-lee was only talking to @StephenCoady about having ionic and cordova installed globally, will get on changing that now.

ciaranRoche commented:
@wei-lee I have added `cordova` and `ionic` to the `package.json`, I have also updated our ios script. I added the node version to the `engines` tag in the `package.json`, but it appears that using `engineStrict` is not used any more according to the docs [here](https://docs.npmjs.com/files/package.json#enginestrict), is there a way around this or is just specifying it in the `package.json` enough?

wei-lee commented:
@ciaranRoche It is enough to just specify it in package.json file. I tried it again locally and it works now without installing ionic and cordova globally.
